sql/parser/builtins.go

```diff
@@ -443,6 +422,171 @@ var builtins = map[string][]builtin{
 	"max": aggregateImpls(boolType, intType, floatType, stringType),
 	"min": aggregateImpls(boolType, intType, floatType, stringType),
 	"sum": aggregateImpls(intType, floatType),
+
+	// Math functions
+
+	"abs": {
+		builtin{
+			returnType: DummyFloat,
+			types:      typeList{floatType},
+			fn: func(args DTuple) (Datum, error) {
+				return DFloat(math.Abs(float64(args[0].(DFloat)))), nil
+			},
+		},
+		builtin{
+			returnType: DummyInt,
+			types:      typeList{intType},
+			fn: func(args DTuple) (Datum, error) {
+				x := args[0].(DInt)
+				switch {
+				case x == math.MinInt64:
+					return x, errAbsOfMinInt64
+				case x < 0:
+					return -x, nil
+				}
+				return x, nil
+			},
+		},
+	},
+
+	"acos": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(math.Acos(x)), nil
+		}),
+	},
+
+	"asin": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(math.Asin(x)), nil
+		}),
+	},
+
+	"atan": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(math.Atan(x)), nil
+		}),
+	},
+
+	"atan2": {
+		floatBuiltin2(func(x, y float64) (Datum, error) {
+			return DFloat(math.Atan2(x, y)), nil
+		}),
+	},
+
+	"ceil":    {ceilImpl},
+	"ceiling": {ceilImpl},
+
+	"cos": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(math.Cos(x)), nil
+		}),
+	},
+
+	"degrees": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(180.0 * x / math.Pi), nil
+		}),
+	},
+
+	"div": {
+		floatBuiltin2(func(x, y float64) (Datum, error) {
+			return DFloat(x / y), nil
+		}),
+	},
+
+	"exp": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(math.Exp(x)), nil
+		}),
+	},
+
+	"floor": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(math.Floor(x)), nil
```

Comment:
It's surprising because I see these functions as a kind of cast: I want an integer, and I'm specifying how I want that integer created. `floor(x)::int` feels redundant, but I guess there's plenty of precedent for doing it this way (Python's `math.floor()` and friends do return integers)